aptos-move/e2e-move-tests/src/harness.rs

```diff
@@ -797,6 +797,51 @@ macro_rules! assert_abort {
     }};
 }
 
+/// Helper to assert transaction aborts.
+#[macro_export]
+macro_rules! assert_abort_ref {
+    // identity needs to be before pattern (both with and without message),
+    // as if we pass variable - it matches the pattern arm, but value is not used, but overriden.
+    // Opposite order and test_asserts_variable_used / test_asserts_variable_used_with_message tests
+    // would fail
+    ($s:expr, $c:ident $(,)?) => {{
+        claims::assert_matches!(
```

Comment:
without using assert_matches (but using assert!(matches!())) we don't get nice error messages.

but assert_matches takes ownership of the object, so we need it to be on references.
will see to dedup on main

Reply:
Should assert_abort always match the references by default?

Reply:
yup, but callers need to be changed, we have a lot of them.